backend-app/middlewares/globalErrorHandler.js

```diff
@@ -11,13 +13,22 @@ const currentEnv = process.env.NODE_ENV || "development";
  * Express automatically knows that this entire function is an error handling middleware by specifying 4 parameters
  */
 module.exports = (err, req, res, next) => {
-  Logger.error(err.stack);
+  // Logger.error(err.stack);
   if (err.code === 11000) {
     Object.keys(err.keyValue).forEach((key) => {
       err.message = `Duplicate value for the field : [${key}] Please use another value!`;
     });
     err.statusCode = 409;
   }
+  // console.log(err);
+  if (err.name === 'ValidationError') {
+      err = new AppError(400, 'fail', err.message);
+  }
+  // If the error dosent have a status code, set it to 500
```

Comment:
We can check if the error is a  instance of our AppError class. Then we can easily filter our api specific errors and handle them seperately .And the other errors will be handled with 500 - internal server error

Reply:
actually we do that in the global middleware error.

```js
const sendErrorProd = (err, res) => {
  if (err.isOperational) {
    res.status(err.statusCode).json({
      status: err.status,
      message: err.message,
      errors: err.errors,
    });
  } else {
    res.status(500).json({
      status: "error",
      message: "Something went very wrong!",
    });
  }
};
```

@muttaqin1

Reply:
I think this is the best way to handle errors. I filter api specific errors using this function in my project:
```ts
export const IsApiError = (err: unknown): err is ApiError =>
  err instanceof ApiError ? err.isOperationalError : false;
```
@Khalid1G

Reply:
can you make a pr with those changes @muttaqin1

Reply:
yeah sure.
@bellaabdelouahab